### ISSUE TITLE
what should the resutls be like if I set the validation set and the prediction set the same?

### ISSUE THREAD

**Issue by o0windseed0o** [faneshion/MatchZoo#38]
A problem occured to me that when I set the training, validation and prediction dataset the same and run the pipline train and predict.

 I used matchpyramid as the matching model, 0.0001 learning rate with 400 epochs for training. It displayed that during the last iterations, the accuracy on the eval set had reached 0.99+. However, when running prediction, loading the model of the last iteration, and predict on the training set, the accuracy was about 0.8. 

Anyone knows what's wrong?

**Comment by uduse** [comment]
If you figured your own question out, don't just erase your description and close the issue. Try answering it as a comment, so people with the same question can be help, also, there's a chance that you answer is not actually correct, so people can check it for you.

**Comment by o0windseed0o** [comment]
@uduse Sorry for that. I'll pay attention to it the next time. Actually I haven't prepared all the evidence and data I need to post this issue. Once ready, I may reopen this issue.

**Comment by yangliuy** [comment]
@o0windseed0o Could you clarify your question a bit more? 

"It displayed that during the last iterations, the accuracy on the **eval set** had reached 0.99+. However, when running prediction, loading the model of the last iteration, and predict on the **training set**, the accuracy was about 0.8."

If you use the same model to do prediction on two different data sets (test and train), the accuracy could be different.

**Comment by o0windseed0o** [comment]
@yangliuy I apologize for the confusion on eval and valid. 
During the training and testing process, I kept the relation file paths for train, valid, and predict the same. The display was something like this:
...
[Train:train] Iter:400 loss=0.0076
[Eval:valid] Iter: 400 accuracy=0.99651

I my view, the second line should be the accuracy on the valid set, right?
But when prediction on the same dataset, I was not able to generate a high accuracy like 0.99.

**Comment by danner1** [comment]
I see, thank you
